src/ngx_http_lua_uri.c

```diff
@@ -55,34 +56,38 @@ ngx_http_lua_ngx_req_set_uri(lua_State *L)
         return luaL_error(L, "attempt to use zero-length uri");
     }
 
-    if (ngx_http_lua_check_unsafe_string(r, p, len, "uri") != NGX_OK) {
-        return luaL_error(L, "attempt to set unsafe uri");
-    }
-
-    if (n == 2) {
-
+    if (n >= 2) {
         luaL_checktype(L, 2, LUA_TBOOLEAN);
         jump = lua_toboolean(L, 2);
+    }
+
+    if (n >= 3) {
+        luaL_checktype(L, 3, LUA_TBOOLEAN);
+        binary = lua_toboolean(L, 3);
+    }
 
-        if (jump) {
+    if (!binary 
+        && ngx_http_lua_check_unsafe_string(r, p, len, "uri") != NGX_OK) {
```

Comment:
style: `{` should move the next line.

Reply:
in othersource code， { are keep on the same line